hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashPolicyOzone.java

```diff
@@ -286,6 +287,10 @@ protected class Emptier implements Runnable {
           trashEmptierCorePoolSize, 1,
           TimeUnit.SECONDS, new ArrayBlockingQueue<>(1024),
           new ThreadPoolExecutor.CallerRunsPolicy());
+
+      // Set a custom thread name with the provided prefix
+      Thread.currentThread()
+          .setName(threadNamePrefix + "-" + Thread.currentThread().getName());
```

Comment:
I don't think we should update the current thread name here.  `Emptier` is a thread, but it is being created in another thread.  We should not change that thread's name here.

To help explain with some extra log:

```
[main] INFO  om.TrashPolicyOzone (TrashPolicyOzone.java:<init>(292)) - ZZZ updating thread name main -> om1--main
[om1--main] INFO  om.TrashPolicyOzone (TrashPolicyOzone.java:<init>(292)) - ZZZ updating thread name om1--main -> om1--om1--main
[om1--om1--main] INFO  om.TrashPolicyOzone (TrashPolicyOzone.java:<init>(292)) - ZZZ updating thread name om1--om1--main -> om1--om1--om1--main
[om1--om1--om1--main] INFO  om.TrashPolicyOzone (TrashPolicyOzone.java:<init>(292)) - ZZZ updating thread name om1--om1--om1--main -> om1--om1--om1--om1--main
```

However, we should provide a `ThreadFactory` with thread name format (e.g. `threadNamePrefix + "TrashEmptier"`) to the `executor` a few lines above.

https://github.com/apache/ozone/blob/02d3f30030f0b95e430de366d6be8af9022d68bf/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashPolicyOzone.java#L286-L289

Also, prefix already has `-` (unless it's empty).

Reply:
Thanks for the help @adoroszlai , I have implemented a ThreadFactory for the executor.
And what Test did we run to get the generate logs you showed above?

Reply:
> what Test did we run to get the generate logs you showed above?

I ran `TestOzoneFileSystem` (but you can run any other that has trash enabled).  Note I added the "updating thread name" log message temporarily, so you won't see it.  But you would see other log messages from the renamed thread.

Reply:
Thanks the thread names seem ok now after the changes 
```
2023-10-17 12:36:10,858 [om1-TrashEmptier] ERROR om.TrashPolicyOzone (TrashPolicyOzone.java:run(352)) - Error attempting to shutdown
```